driver-core/src/test/functional/com/mongodb/client/model/AggregatesTest.java

```diff
@@ -18,26 +18,120 @@
-
-import org.bson.conversions.Bson;
-import org.junit.jupiter.api.Test;
+import java.util.stream.Stream;
 
 import static com.mongodb.ClusterFixture.serverVersionAtLeast;
+import static com.mongodb.client.model.Accumulators.median;
+import static com.mongodb.client.model.Accumulators.percentile;
 import static com.mongodb.client.model.Aggregates.geoNear;
+import static com.mongodb.client.model.Aggregates.group;
 import static com.mongodb.client.model.Aggregates.unset;
 import static com.mongodb.client.model.GeoNearOptions.geoNearOptions;
+import static com.mongodb.client.model.Sorts.ascending;
+import static com.mongodb.client.model.Windows.Bound.UNBOUNDED;
+import static com.mongodb.client.model.Windows.documents;
 import static java.util.Arrays.asList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assumptions.assumeTrue;
 
 public class AggregatesTest extends OperationTest {
 
+    private static Stream<Arguments> groupWithQuantileSource() {
+        return Stream.of(
+                Arguments.of(percentile("sat_95", "$x", new double[]{0.95}, QuantileMethod.approximate()), asList(3.0), asList(1.0)),
+                Arguments.of(percentile("sat_95", "$x", new double[]{0.95, 0.3}, QuantileMethod.approximate()), asList(3.0, 2.0),
+                        asList(1.0, 1.0)),
+                Arguments.of(median("sat_95", "$x", QuantileMethod.approximate()), 2.0d, 1.0d)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("groupWithQuantileSource")
+    public void shouldGroupWithQuantile(final BsonField quantileAccumulator,
+                                        final Object expectedGroup1,
+                                        final Object expectedGroup2) {
+        //given
+        assumeTrue(serverVersionAtLeast(7, 0));
+        Document[] original = new Document[]{
+                new Document("x", 1).append("z", false),
+                new Document("x", 2).append("z", true),
+                new Document("x", 3).append("z", true)
+        };
+        getCollectionHelper().insertDocuments(original);
+
+        //when
+        List<Document> results = getCollectionHelper().aggregate(Collections.singletonList(
+                group(new Document("gid", "$z"), quantileAccumulator)), DOCUMENT_DECODER);
+
+        //then
+        assertThat(results, hasSize(2));
+
+        Object result = results.stream()
+                .filter(document -> document.get("_id").equals(new Document("gid", true)))
+                .findFirst().map(document -> document.get("sat_95")).get();
 
+        assertEquals(expectedGroup1, result);
+
+        result = results.stream()
+                .filter(document -> document.get("_id").equals(new Document("gid", false)))
+                .findFirst().map(document -> document.get("sat_95")).get();
+
+        assertEquals(expectedGroup2, result);
```

Comment:
It seems that the above boilerplate could be replaced with, for example:
```
        assertResults(pipeline, "[\n"
                + "   { _id: true, sat_95: [3, 2] },\n"
                + "   { _id: false, sat_95: [1, 2] }\n"
                + "]");
```
This seems easier to follow than using a stream to pick out the ids, and then manually matching on the particular field.

So instead of the ~10 lines of boilerplate, plus these ~3 lines per test:

```
Arguments.of(
    percentile("sat_95", "$x", new double[]{0.95}, QuantileMethod.approximate()),
    asList(3.0), asList(1.0)),
```

there could be ~6 lines per test:

```
        List<Bson> pipeline = assertPipeline(
                ...
                group("$z", percentile("sat_95", "$x", ofNumberArray(0.95))));

        assertResults(pipeline, "[\n"
                + "   { _id: true, sat_95: [3, 2] },\n"
                + "   { _id: false, sat_95: [1, 1] }\n"
                + "]");
```

I count the lines because parameterization is often used to decrease boilerplate and line count, but to me it seems there can often be other ways to deal with that problem which ultimately make the code more clear.

I think it is fine to have a some duplication in the "data" part of a test (for example, trying to remove it separates `percentile` from `group`, and the array values from `sat_95`, which makes things harder to follow).

Reply:
Discussed: we can address this later, in a future PR (or, decide not to). We should have a consistent style in the tests.